config/routes.rb

```diff
@@ -1,3 +1,21 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  scope module: :v1, constraints: ApiVersion.new('v1', true) do
+    resources :projects do
+      resources :tasks
+    end
+
+    resources :tasks do
+      resources :comments
+    end
+  end
+
+  scope module: :v2, constraints: ApiVersion.new('v2') do
+    resources :projects, only: :index
+  end
+
+
+  post 'auth/sign_in', to: 'authentication#authenticate'
```

Comment:
need to add delete for log out

Reply:
as I use JWT token I can't create path for logout, for log out responsible this code:
def self.encode(payload, exp = 24.hours.from_now)
    payload[:exp] = exp.to_i
    JWT.encode(payload, HMAC_SECRET)
  end
he's in app/lib/json_web_token.rb

Reply:
that's why I suggested (in skype) to create a new field in `User` table and put the token there on login, and on the logout you would just erase the token from the user's record